### PR TITLE
Coverage increase in injector_helper (resolves #44)

### LINKED ISSUE #44: coverage for injector_helper
To better understand what we'll need to do to get good test coverage, let's start with injector helper.

### PR DESCRIPTION
**Pull request by inetic**

### CODE REVIEW

**Comment by inetic** [comment]
Pls, don't merge yet. I'm not sure I understand the coverage results.

File: injector.c [hunk 1]
```diff
@@ -63,9 +63,10 @@ int get_port_for_scheme(const char *scheme)
     hints.ai_socktype = SOCK_STREAM;
     hints.ai_flags = AI_PASSIVE;
     struct addrinfo *res;
+    fprintf(stderr, "scheme:%s\n", scheme);
```
Comment (shalunov) [inline injector.c]: Could you avoid changes to injector.c?

File: test.sh [hunk 2]
```diff
@@ -10,9 +10,17 @@ origin_addr=localhost
 origin_port=8080
 injector_tcp_port=8005
 proxy_tcp_port=5678
+GOOGLE=172.217.7.142
```
Comment (shalunov) [inline test.sh]: Why the IP? Google.com and other Google domains resolve is complicated source- and time-dependent ways, sometimes through CNAMEs. If you want google.com, I'd use that directly.

File: test.sh [hunk 3]
```diff
@@ -68,52 +77,59 @@ function test_n {
...
 for i in 1 8 16; do
-    if ! test_n $i $HTTP_OK; then
+    if ! test_n $i $origin_addr:$origin_port $HTTP_OK; then
         r=1; break;
     fi
 done
 
 if [ "$r" == "0" ]; then
+    echo "$(now) Testing HTTPS forwarding."
+    if ! do_curl $proxy_tcp_port $GOOGLE $HTTP_OK; then
```
Comment (shalunov) [inline test.sh]: `s/\$GOOGLE/google.com/`?

File: test.sh [hunk 4]
```diff
@@ -10,9 +10,17 @@ origin_addr=localhost
 origin_port=8080
 injector_tcp_port=8005
 proxy_tcp_port=5678
+GOOGLE=172.217.7.142
 HTTP_OK=200
 HTTP_BAD_GATEWAY=502
 
+trap cleanup SIGINT SIGTERM EXIT
```
Comment (shalunov) [inline test.sh]: Good idea. I had things left hanging after ^C.

**Review by shalunov — CHANGES_REQUESTED** [review]

File: injector.c (hunk 1 above)
Comment (inetic) [reply injector.c]: Yes, this slipped in while debugging, will remove.

File: test.sh (hunk 2 above)
Comment (inetic) [reply test.sh]: Also part of debugging, will remove.

File: test.sh (hunk 4 above)
Comment (inetic) [reply test.sh]: Yes, or after a test failed, was quite annoying.

File: test.sh (hunk 3 above)
Comment (inetic) [reply test.sh]: yip

**Comment by inetic** [comment]
OK, this one is ready to be merged now. (resolves #44)

**Review by shalunov — APPROVED** [review]

**Pull request merged**